src/register/register-gnome/gnucash-sheet-private.c

```diff
@@ -423,42 +423,41 @@ draw_cell (GnucashSheet *sheet,
     /* top */
     draw_cell_line (cr, bg_color,
                     (borders.top >= borders.left ? x : x + 1.0),
-                    y + 0.5,
+                    (y == 0 ? y + 0.5 : y - 0.5),
```

Comment:
The lost double line is due to another commit. When you reworked the cursor drawing, you apparently removed the extra line drawn there...